File: leetcode/Algorithms/496.NextGreaterElementI/Solution.py
class Solution(object):
    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        return [next((y for y in nums[nums.index(x):] if y > x), -1) for x in findNums]

File: leetcode/Algorithms/496.NextGreaterElementI/Solution.java
import java.util.HashMap;
import java.util.Stack;

public class Solution {
    public int[] nextGreaterElement(int[] findNums, int[] nums) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        Stack<Integer> stack = new Stack<>();

        for (int num : nums) {
            while (!stack.isEmpty() && stack.peek() < num) hashMap.put(stack.pop(), num);
            stack.add(num);
        }

        int[] result = new int[findNums.length];
        for (int k = 0; k < findNums.length; k++) {
            result[k] = hashMap.getOrDefault(findNums[k], -1);
        }
        return result;
    }
}
